Zend/zend_portability.h

```diff
@@ -21,6 +21,12 @@
 #ifndef ZEND_PORTABILITY_H
 #define ZEND_PORTABILITY_H
 
+#ifdef PHP_WIN32
+#include "config.w32.h"
+#else
+#include "php_config.h" // for HAVE_*
```

Comment:
I don't think Zend files included php ones before. Not sure it should be starting now.

Reply:
They did. This include is necessary, or else the config macros are not available, and thus the code could break. Previously, it was included indirectly by somebody else.

Reply:
By design - TSRM doesn't depend on Zend and php, Zend doesn't depend on php, php main doesn't depend on extensions. Unfortunately, this was already violated in few places.

php_confih.h is special and can be used everywhere (I'm not completely sure about TSRM).

Reply:
What bothers me about php_config.h is that you always need to ifdef ZEND_WIN32 and include a different header on Windows. I think there should only be one header that makes config macros available.